src/k8s/pkg/k8sd/images/images.go

```diff
@@ -0,0 +1,22 @@
+package images
+
+import "slices"
+
+var registeredImages []string
+
+// Images returns the list of images registered by individual components.
+func Images() []string {
+	if registeredImages == nil {
+		return nil
```

Comment:
Any specific reason why you return nil here instead of an empty list?

Reply:
it is customary to return `nil` instead of an empty list when the result is empty. Go will handle it in the same way (range over nil does not do anything)